shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -28,62 +29,141 @@
+  // Meta-data arguments
   // TODO: where did this package path come from? is this what it should be?
   private static final String SPLASH_SCREEN_META_DATA_KEY = "io.flutter.app.android.SplashScreenUntilFirstFrame";
-  private static final String EXTRA_INITIAL_ROUTE = "route";
   // TODO: where did this package path come from? is this what it should be?
   private static final String INITIAL_ROUTE_META_DATA_KEY = "io.flutter.app.android.InitialRoute";
 
-  // FlutterFragment identifiers within this Activity.
+  // Intent extra arguments
+  private static final String EXTRA_SHOW_SPLASH_SCREEN = "show_launch_screen";
+  private static final String EXTRA_DART_ENTRYPOINT = "dart_entrypoint";
+  private static final String EXTRA_INITIAL_ROUTE = "route";
+
+  // FlutterFragment management
   private static final String TAG_FLUTTER_FRAGMENT = "flutter_fragment";
-  private static final int CONTAINER_ID = 609893468; // random number
+  private static final int FRAGMENT_CONTAINER_ID = 609893468; // random number
   private FlutterFragment flutterFragment;
 
   @Override
   public void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    Log.d(TAG, "onCreate()");
+    setContentView(createFragmentContainer());
+    ensureFlutterFragmentCreated();
+    configureStatusBarColor();
+  }
 
-    FlutterShellArgs args = FlutterShellArgs.fromIntent(getIntent());
-    // TODO(mattcarroll): Change FlutterMain to accept FlutterShellArgs and move additional constants in
-    //       FlutterMain over to FlutterShellArgs.
-    FlutterMain.ensureInitializationComplete(getApplicationContext(), args.toArray());
+  /**
+   * Sets the color of this {@code Activity}'s status bar to a translucent black, i.e., a dark
+   * overlay.
+   *
+   * Beginning with Android Lollipop, the status bar at the top of the screen can have its color
+   * set at runtime.
+   */
+  @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+  private void configureStatusBarColor() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      Window window = getWindow();
+      window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+      window.setStatusBarColor(0x40000000);
+      window.getDecorView().setSystemUiVisibility(PlatformPlugin.DEFAULT_SYSTEM_UI);
+    }
+  }
 
-    configureStatusBarColor();
-    createFlutterFragmentContainer();
-    createFlutterFragment();
+  /**
+   * If no {@code FlutterFragment} exists in this {@code FlutterActivity}, then a {@code FlutterFragment}
+   * is created and added. If a {@code FlutterFragment} does exist in this {@code FlutterActivity}, then
+   * a reference to that {@code FlutterFragment} is retained in {@code #flutterFragment}.
+   */
+  private void ensureFlutterFragmentCreated() {
+    FragmentManager fragmentManager = getSupportFragmentManager();
+    flutterFragment = (FlutterFragment) fragmentManager.findFragmentByTag(TAG_FLUTTER_FRAGMENT);
+
```

Comment:
uber-nit: I'd drop the blank line since the check below is closely associated with the lookup on line 115.